python/cudf/cudf/core/dataframe.py

```diff
@@ -7133,12 +7133,12 @@ def eval(self, expr: str, inplace: bool = False, **kwargs):
             )
 
         # Have to use a regex match to avoid capturing "=="
-        includes_assignment = re.search("[^=]=[^=]", expr) is not None
+        includes_assignment = re.search("[^=><]=[^=><]", expr) is not None
 
         # Check if there were multiple statements. Filter out empty lines.
         statements = tuple(filter(None, expr.strip().split("\n")))
         if len(statements) > 1 and any(
-            re.search("[^=]=[^=]", st) is None for st in statements
+            re.search("[^=><]=[^=><]", st) is None for st in statements
```

Comment:
Given that this is repeated and slightly nontrivial, I’d like to use a helper function or a name for the regex pattern.

Reply:
Done.